packages/components/src/components/tables/es-table/es-table.css

```diff
@@ -10,3 +10,18 @@ es-table {
 }
 
 @import url('../table.css');
+
+.loading {
+    opacity: 0;
+    animation: fadeIn 2s forwards;
+    animation-delay: 0.2s;
+}
+
+@keyframes fadeIn {
+    from {
+        opacity: 0;
+    }
+    to {
+        opacity: 1;
+    }
+}
```

Comment:
Let's move this up to `Page`, on it's `main`. Adding it only to the table means that we will still get a flash of the page title, etc, and also means it needs to be added in multiple places, especially if we were to add the loading to non-table based screens.

Reply:
Let me figure out how to share the table `loading` state to the Page component effectively. I’ll explore the following approaches:

1. Event-Driven State Management:
   - Emit custom events to notify the parent component or document.body about loading state changes.
   - Aggregate these states in a centralized place to determine the overall loading status.
2. Global State Management:
   - Use a simple global state management solution to track the loading state.
   - Make sure the parent page element reacts to changes in this global state.

Reply:
`Page` already has the page state (including loading).

https://github.com/EventStore/Design-System/blob/main/packages/layout/src/components/Page/Page.tsx#L140

```diff
export const Page: FunctionalComponent<PageProps> = (props, children) => (
    <>
        <PageTitle>{props.pageTitle}</PageTitle>
        <main 
            ref={(ref) => ((document as any).main = ref)}
+           class={{ loading: props.state === "loading" }} 
        >
            {props.crumbs && <es-breadcrumb crumbs={props.crumbs} />}
            <PageBody {...props}>{children}</PageBody>
        </main>
        <PageView title={props.pageTitle} />
    </>
);
```

https://github.com/EventStore/Design-System/blob/main/packages/layout/src/css/page.css#L25

```diff
    height: 100%;
    padding: 30px 40px;
+
+    &.loading {
+        opacity: 0;
+        animation: fadeIn 2s forwards;
+        animation-delay: 0.2s;
+    }
}

+@keyframes fadeIn {
+    from {
+        opacity: 0;
+    }
+    to {
+        opacity: 1;
+    }
+}
```

Reply:
![2024-06-24 at 17 38 59 - Gold Macaw](https://github.com/EventStore/Design-System/assets/25894545/45234f90-c544-41ef-af21-2560c3b72bc0)

Reply:
Cool, looking at the timings in the cloud UI, I think we could bump the delay up to `400` and reduce the fade to our standard `500`. 

We also need to set the duration to `0` for `prefers-reduced-motion` users.

Reply:
![2024-06-24 at 23 39 57 - Gray Falcon](https://github.com/EventStore/Design-System/assets/25894545/c8e8a49b-a1d5-401c-8e08-13e6a8d7e641)

Reply:
![2024-06-24 at 23 44 18 - Indigo Peafowl](https://github.com/EventStore/Design-System/assets/25894545/a1dd6791-c2d8-47b2-af6b-4e63a6963326)